Source/Core/VideoCommon/UberShaderPixel.cpp

```diff
@@ -384,285 +386,285 @@ ShaderCode GenPixelShader(APIType ApiType, const ShaderHostConfig& host_config,
       "    return (color_A.r == color_B.r && color_A.g == color_B.g && color_A.b == color_B.b);\n"
       "  default:\n"
       "    return false;\n"
-      "  }\n"
-      "}\n\n");
+      "  }}\n"
+      "}}\n\n");
 
   // =================
   //   Input Selects
   // =================
 
-  out.Write("struct State {\n"
-            "  int4 Reg[4];\n"
-            "  int4 TexColor;\n"
-            "  int AlphaBump;\n"
-            "};\n"
-            "struct StageState {\n"
-            "  uint stage;\n"
-            "  uint order;\n"
-            "  uint cc;\n"
-            "  uint ac;\n");
-
-  out.Write("};\n"
-            "\n"
-            "int4 getRasColor(State s, StageState ss, float4 colors_0, float4 colors_1);\n"
-            "int4 getKonstColor(State s, StageState ss);\n"
-            "\n");
+  out.WriteFmt("struct State {{\n"
+               "  int4 Reg[4];\n"
+               "  int4 TexColor;\n"
+               "  int AlphaBump;\n"
+               "}};\n"
+               "struct StageState {{\n"
+               "  uint stage;\n"
+               "  uint order;\n"
+               "  uint cc;\n"
+               "  uint ac;\n");
+
+  out.WriteFmt("}};\n"
```

Comment:
This looks really weird, with the `struct StageState` opening in the previous call and closing in this one. Is this a left-over of some old code that generated dynamic members?

It would probably look nicer if you moved the curly over, combined all into one `WriteFmt` call, or split them up into three `WriteFmt` calls (one for `struct State`, one for `struct StageState` and one for the...forward declarations of functions? - each with their respective curlies)